runners/s3-benchrunner-c/BenchmarkRunner.cpp

```diff
@@ -218,20 +219,52 @@ void printAllStats(uint64_t bytesPerRun, const vector<double> &durations)
     printf("Peak RSS:%f MiB\n", (double)mu.maxrss / 1024.0);
 }
 
+struct Args
+{
+    string s3ClientId;
+    string workload;
+    string bucket;
+    string region;
+    double targetThroughputGbps;
+
+    // Optional arguments
+    string network_interface_names = "";
+};
+
 int benchmarkRunnerMain(int argc, char *argv[], const CreateRunnerFromNameFn &createRunnerFromName)
 {
-    if (argc != 6 && argc != 7)
-        fail(string("usage: ") + argv[0] + " S3_CLIENT WORKLOAD BUCKET REGION TARGET_THROUGHPUT [NETWORK_INTERFACES]");
-
-    string s3ClientId = argv[1];
-    string workload = argv[2];
-    string bucket = argv[3];
-    string region = argv[4];
-    double targetThroughputGbps = stod(argv[5]);
-    string network_interface_names = (argc == 7) ? argv[6] : "";
-
-    auto config = BenchmarkConfig(workload, bucket, region, targetThroughputGbps, network_interface_names);
-    unique_ptr<BenchmarkRunner> benchmark = createRunnerFromName(s3ClientId, config);
+    // START Argument Parsing
+    argh::parser cmdl;
+    // pre-register optional named arguments to support --param_name param_value syntax
+    cmdl.add_params({"--nic"});
+    cmdl.parse(argc, argv);
+
+    if (cmdl[{"-h", "--help"}] || cmdl.pos_args().size() < 6)
+    {
+        fail(
+            std::string("usage: ") + argv[0] + " S3_CLIENT WORKLOAD BUCKET REGION TARGET_THROUGHPUT --nic name1,name2");
```

Comment:
`[` braces to indicate optional 
```suggestion
            std::string("usage: ") + argv[0] + " S3_CLIENT WORKLOAD BUCKET REGION TARGET_THROUGHPUT [--nic name1,name2]");
```